third_party/amd/include/hipblas_instance.h

```diff
@@ -0,0 +1,277 @@
+
+  void successOrExit(hipblasStatus_t status, const std::string &context = "") {
+    if (status != HIPBLAS_STATUS_SUCCESS) {
+      std::string error_msg =
+          "HIPBLAS Error in " + context + ": " + std::to_string(status);
+
+      switch (status) {
+      case HIPBLAS_STATUS_NOT_INITIALIZED:
+        error_msg += " (NOT_INITIALIZED)";
+        break;
+      case HIPBLAS_STATUS_ALLOC_FAILED:
+        error_msg += " (ALLOC_FAILED)";
+        break;
+      case HIPBLAS_STATUS_INVALID_VALUE:
+        error_msg += " (INVALID_VALUE - Parameters are unexpectedly NULL, in "
+                     "conflict or in impossible configuration)";
+        break;
+      case HIPBLAS_STATUS_MAPPING_ERROR:
+        error_msg += " (MAPPING_ERROR)";
+        break;
+      case HIPBLAS_STATUS_EXECUTION_FAILED:
+        error_msg += " (EXECUTION_FAILED)";
+        break;
+      case HIPBLAS_STATUS_INTERNAL_ERROR:
+        error_msg += " (INTERNAL_ERROR)";
+        break;
+      case HIPBLAS_STATUS_NOT_SUPPORTED:
+        error_msg += " (NOT_SUPPORTED)";
+        break;
+      case HIPBLAS_STATUS_INVALID_ENUM:
+        error_msg +=
+            " (INVALID_ENUM - unsupported enum value was passed to function)";
+        break;
+      case HIPBLAS_STATUS_UNKNOWN:
+        error_msg += " (UNKNOWN)";
+        break;
+      default:
+        error_msg += " (UNKNOWN_ERROR_CODE)";
+      }
+
+      throw std::runtime_error(error_msg + "\n");
+    }
+  }
+
+  void gemm_impl(int m, int n, int k, uint64_t A, uint64_t B, uint64_t C,
+                 uint64_t D, hipDataType dtype, float alpha, float beta) {
+
+    hipblasLtMatmulDesc_t matmulDesc = NULL;
+
+    hipblasOperation_t transa = HIPBLAS_OP_T;
+    hipblasOperation_t transb = HIPBLAS_OP_N;
+
+    int8_t fastAccum = 1;
+
+    hipblasLtMatrixLayout_t Adesc = NULL, Bdesc = NULL, Cdesc = NULL,
+                            Ddesc = NULL;
+
+    int returnedResults = 0;
+    hipblasLtMatmulHeuristicResult_t heuristicResult = {};
+
+    hipblasComputeType_t computeType;
+    if (dtype == HIP_R_32F) {
+      computeType = HIPBLAS_COMPUTE_32F_FAST_TF32;
+    } else if (dtype == HIP_R_8F_E4M3_FNUZ) {
+      computeType = HIPBLAS_COMPUTE_32F_FAST_8F_FNUZ;
+    } else {
+      computeType = HIPBLAS_COMPUTE_32F;
+    }
+    successOrExit(
+        hipblasLtMatmulDescCreate(&matmulDesc, computeType, HIP_R_32F));
+    successOrExit(hipblasLtMatmulDescSetAttribute(
+        matmulDesc, HIPBLASLT_MATMUL_DESC_TRANSA, &transa, sizeof(transa)));
+    successOrExit(hipblasLtMatmulDescSetAttribute(
+        matmulDesc, HIPBLASLT_MATMUL_DESC_TRANSB, &transb, sizeof(transb)));
+
+    auto c_dtype = dtype == HIP_R_8F_E4M3_FNUZ ? HIP_R_16F : dtype;
+
+    successOrExit(hipblasLtMatrixLayoutCreate(&Adesc, dtype, k, m, k));
+    successOrExit(hipblasLtMatrixLayoutCreate(&Bdesc, dtype, k, n, k));
+    successOrExit(hipblasLtMatrixLayoutCreate(&Cdesc, c_dtype, m, n, m));
+    // FP8 inputs require FP16 output layout for valid heuristics?
```

Comment:
All are accumulating into f32 so I'm not sure this is the case for our side. Maybe NVIDIA specific?